frontend/public/components/namespace.jsx

```diff
@@ -831,43 +837,54 @@ export const ProjectsPage = connectToFlags(
 /** @type {React.SFC<{namespace: K8sResourceKind}>} */
 export const PullSecret = (props) => {
   const [isLoading, setIsLoading] = React.useState(true);
-  const [data, setData] = React.useState(undefined);
+  const [data, setData] = React.useState([]);
+  const [error, setError] = React.useState(false);
   const { t } = useTranslation();
+  const { namespace, canViewSecrets } = props;
 
   React.useEffect(() => {
-    k8sGet(SecretModel, null, props.namespace.metadata.name, {
-      queryParams: { fieldSelector: 'type=kubernetes.io/dockerconfigjson' },
-    })
-      .then((pullSecrets) => {
+    k8sGet(ServiceAccountModel, 'default', namespace.metadata.name, {})
+      .then((serviceAccount) => {
         setIsLoading(false);
-        setData(_.get(pullSecrets, 'items[0]'));
+        setData(serviceAccount.imagePullSecrets ?? []);
+        setError(false);
       })
-      .catch((error) => {
+      .catch(() => {
         setIsLoading(false);
-        setData(undefined);
-        // A 404 just means that no pull secrets exist
-        if (error.status !== 404) {
-          throw error;
-        }
+        setData([]);
+        setError(true);
```

Comment:
Worth logging the error details with `console.error` here for debugging.

Reply:
Add console.error